internal/backend/gs/gs.go

```diff
@@ -301,16 +317,45 @@ func (be *Backend) openReader(ctx context.Context, h restic.Handle, length int,
 
 	objName := be.Filename(h)
 
+	if be.encryptKey != "" {
+		encryptKeyPresent = true
+	}
+	for _, key := range be.decryptKeys {
+		if key != "" {
+			decryptKeysPresent = true
+			break
+		}
+	}
+
 	be.sem.GetToken()
 
-	r, err := be.bucket.Object(objName).NewRangeReader(ctx, offset, int64(length))
+	//https://cloud.google.com/storage/docs/encryption/using-customer-supplied-keys
+	err = errors.New(("Google Storage customer-supplied encryption keys failed."))
+	if encryptKeyPresent {
```

Comment:
Why not use `if be.encryptKey != "" {` here?

Reply:
Please just add the encryption key to the list of decryption keys. That way we don't need separate code paths.

Reply:
Due to the use of a map in the latest version, I can't lump all keys together.